Fix/gatang date feature

JirayuNam opened the pull request:
# Pull Request Template

## Description

*Change every date format to Buddhism Calendar and prepare function for exporting date text.*

- **Type of Change:** Bug fix / New feature /

## Changes Made

*List out the specific changes or updates made in this PR.*

- [ ] date-feature.tsx: DateText function, prepare JSX export to use and showing for needed component
- [ ] date-feature.tsx: BuddhistGregor function, format date into Buddhist Gregorian Calendar
- [ ] date-featyre.tsx: DatePickerWithRange, change format caption on shadcn calendar to Buddhism (return). Set formattedDate state to change text inside
- [ ] date-featyre.tsx: DatePickerWithRangeProject, change format caption on shadcn calendar to Buddhism (return). Set formattedDate state to change text inside
- [ ] projectdetail.tsx: change return in const MenuBar to use DatePickerWithRangeProject on line:356
- [ ] taskDate.tsx: delete formatDate and use DateText instead

## How to Test

*Provide step-by-step instructions on how to verify the changes. Include any setup details or special conditions that need to be met.*

1. DateTest using on Home's page on project card and Task Management's page on each task line.
2. DatePicker test by picking 3 condition: 1. No date selected, 2. No endDate, 3. have both start and end date
3. DatePicker: Check DB if sending data correctly.

## Screenshots (if applicable)

*Attach any screenshots or screen recordings that demonstrate the new feature, UI changes, or bug fix.*

## Checklist

*Please ensure that all boxes are checked (replace [ ] with [x]). This checklist will help in maintaining a high standard for each PR.*

- [ ] I have tested the changes and they work as expected.
- [ ] I have added relevant documentation (if applicable).
- [ ] I have updated relevant tests (if applicable).
- [ ] This PR has a descriptive title and description.
- [ ] This PR follows the code style and conventions of the project.

## Additional Notes

*Add any other context or information that reviewers should be aware of.*

---

*Thank you for your contribution! We appreciate your effort to make this project better.*

Romeliette reviewed (changes requested):
files changed 55 files เลยอ่ะเหมือนมันจะไปเอา commit ของเกมมาด้วย เปลี่ยนเป็น rebase แทนนะครับคือเอาแค่ commit ล่าสุดมา

JirayuNam committed:
fix(buddhist date + date format function): all year to buddhist year + create date format func

Change all year involve in date-feature to buddhist date year and create date format func for other
to use,

BREAKING CHANGE: projectdetail.tsx on MenuBar add project state and on date format change to
DatePickerWithRangeProject + taskDate change fromd formatDate to DateText.
